### PR TITLE
Chore/ci/rework

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

Speed up CI, use cache more efficiently, and organize the workflows better.

## Links to any relevant issues

- #91 
- #165

### CODE REVIEW

**Comment by DaughterOfMars** [comment]
TODO: Remove coverage from this branch before merging

File: .github/workflows/bindings-wallet-publish-java.yml
Comment (thibault-martinez) [inline .github/workflows/bindings-wallet-publish-java.yml:1]: Can you put the publish part at the end so the order group them by lang please?

File: .github/workflows/bindings-client-wasm.yml [hunk 1]
```diff
@@ -1,39 +1,41 @@
...
       - ".github/actions/**"
       - "**.rs" # Include all rust files
       - "**Cargo.toml" # Include all Cargo.toml files
       - "**Cargo.lock" # Include all Cargo.lock files
       - "!**/examples/**" # Exclude all examples
       - "!**/tests/**" # Exclude all tests
-      - "!**/bindings/**" # Exclude all bindings
       - "!cli/**" # Exclude CLI
-      - "sdk/src/client/bindings/wasm/**" # Re-include wasm client bindings
-      - "sdk/src/client/bindings/nodejs/**" # Re-include node client bindings, b/c lib copied from there
```
Comment (Thoralf-M) [inline .github/workflows/bindings-client-wasm.yml:16]: Nodejs binding should stay, also below on pull_requests and should also be added for the wallet binding (there the wallet nodejs binding ofc)

**Review by Thoralf-M — APPROVED** [review]

**Review by Thoralf-M — DISMISSED** [review]

File: .github/workflows/publish-cli.yml
Comment (thibault-martinez) [inline .github/workflows/publish-cli.yml:1]: I'd put publish at the end for consistency

File: Cargo.toml [hunk 2]
```diff
@@ -17,9 +17,16 @@ members = [
 
 [profile.dev]
 panic = "abort"
-# Disabling debug info speeds up builds a bunch,
-# and we don't rely on it for debugging that much.
+
+# Necessary to get coverage to work as for some reason using a custom profile breaks it
+[profile.test]
```
Comment (thibault-martinez) [inline Cargo.toml:22]: Is this what you were annoyed with?

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -188,14 +137,17 @@ jobs:
...
           install: cargo-audit
-          cache: true
-          cache-job-id: ${{ github.workflow }}-${{ github.job }}
-          cache-hash: ${{ hashFiles('.github/workflows/core.yml') }}
+
+      - name: Install cargo-audit
+        uses: actions-rs/cargo@v1
+        with:
+          command: install
+          args: cargo-audit
```
Comment (thibault-martinez) [inline .github/workflows/lint.yml]: Isn' the previous command with `install: cargo-audit` doing that already?

File: .github/workflows/coverage.yml [hunk 4]
```diff
@@ -2,7 +2,7 @@ name: Test coverage
 
 on:
   push:
-    branches: [develop, production]
+    branches: [develop, production, chore/ci/rework]
```
Comment (thibault-martinez) [inline .github/workflows/coverage.yml]: Remove?

File: Cargo.toml (hunk 2 above)
Comment (DaughterOfMars) [reply Cargo.toml:22]: Yes

File: .github/workflows/lint.yml (hunk 3 above)
Comment (DaughterOfMars) [reply .github/workflows/lint.yml]: Oops

File: .github/workflows/coverage.yml (hunk 4 above)
Comment (DaughterOfMars) [reply .github/workflows/coverage.yml]: Yes, thank you.

File: Cargo.toml (hunk 2 above)
Comment (thibault-martinez) [reply Cargo.toml:22]: Interesting

**Review by thibault-martinez — APPROVED** [review]

File: .github/workflows/bindings-client-java.yml [hunk 5]
```diff
@@ -1,62 +1,63 @@
...
 jobs:
   test:
     name: Test
-    if: ${{ ! github.event.schedule }}
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
-        os: [windows-latest, macos-latest, ubuntu-latest]
+        # os: [windows-latest, macos-latest, ubuntu-latest]
```
Comment (kwek20) [inline .github/workflows/bindings-client-java.yml:48]: comment here

Comment (DaughterOfMars) [reply .github/workflows/bindings-client-java.yml:48]: You mean add a comment?

**Review by Thoralf-M — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

**Pull request merged**